qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py

```diff
@@ -107,23 +110,34 @@ class LatticeDrawStyle:
 class Lattice:
     """General Lattice."""
 
-    def __init__(self, graph: PyGraph) -> None:
+    def __init__(self, graph: Union[PyGraph, nx.Graph]) -> None:
         """
         Args:
-            graph: Input graph for Lattice. `graph.multigraph` must be False.
+            graph: Input graph for Lattice. Can be provided as ``retworkx.PyGraph``, which is
+                used internally, or, for convenience, as ``networkx.Graph``. The graph
+                cannot be a multigraph.
 
         Raises:
-            ValueError: If `graph.multigraph` is True for a given graph, it is invalid.
+            ValueError: If the input graph is a multigraph.
+            ValueError: If the graph edges are non-numeric.
         """
+        if isinstance(graph, nx.Graph):
+            graph = networkx_converter(graph)
+
         if graph.multigraph:
             raise ValueError(
                 f"Invalid `graph.multigraph` {graph.multigraph} is given. "
                 "`graph.multigraph` must be `False`."
             )
-        if graph.edges() == [None] * graph.num_edges():
-            weighted_edges = [edge + (1.0,) for edge in graph.edge_list()]
-            for start, end, weight in weighted_edges:
-                graph.update_edge(start, end, weight)
+
+        # validate the edge weights
+        for edge_index, weight in enumerate(graph.edges()):
+            if weight is None or weight == {}:
+                # None or {} is updated to be 1
+                graph.update_edge_by_index(edge_index, 1)
+            elif not isinstance(weight, numbers.Number):
+                raise ValueError(f"Unsupported weight {weight} on edge with index {edge_index}.")
```

Comment:
This won't work if the graph has edges removed. In retworkx the edge indices (or node indices) aren't guaranteed to be contiguous as removals will leave holes in the index list (see the description here: https://qiskit.org/documentation/retworkx/stubs/retworkx.PyGraph.html#retworkx.PyGraph ). So if someone passes in a graph with a removed node this will be updating the wrong edge. Instead I would change this to:

```suggestion
        for edge_index, weight in graph.edge_index_map().items():
            if weight is None or weight == {}:
                # None or {} is updated to be 1
                graph.update_edge_by_index(edge_index, 1)
            elif not isinstance(weight, numbers.Number):
                raise ValueError(f"Unsupported weight {weight} on edge with index {edge_index}.")
```

Reply:
Thanks! I've also added a test to cover this case.